### PR TITLE
Adds callback cache identity

### PR DESCRIPTION
**Pull request by vegaro**
We are getting some customers calling alias or identify too often. Sometimes with the same parameters. 

Something I have noticed while trying to replicate the issue with this code:

```kotlin
for (i in 1..10) {
    if (newUserId != Purchases.sharedInstance.appUserID) {
        Purchases.sharedInstance.loginWith {
            // Do other stuff
        }
    }
}
```

Is that most of the times the code will enter in the `if` clause. The reason is that we only update the appUserID when the API call has come back with a result, so the check would most of the times evaluate to the same.

In this PR, I have added the same caching mechanism we currently have for other API calls. It prevents the same calls being executed at the same time with the same parameters. This should prevent apps from hitting our servers with `/alias` and `/identify` calls that look the same too often.

### CODE REVIEW

File: common/src/main/java/com/revenuecat/purchases/common/Backend.kt [hunk 1]
```diff
@@ -48,6 +53,12 @@ class Backend(
     @get:Synchronized @set:Synchronized
     @Volatile var offeringsCallbacks = mutableMapOf<String, MutableList<OfferingsCallback>>()
 
+    @get:Synchronized @set:Synchronized
+    @Volatile var createAliasCallbacks = mutableMapOf<CallbackCacheKey, MutableList<CreateAliasCallback>>()
+
+    @get:Synchronized @set:Synchronized
+    @Volatile var identifyCallbacks = mutableMapOf<CallbackCacheKey, MutableList<IdentifyCallback>>()
+
```
Comment (aboedo) [inline common/src/main/java/com/revenuecat/purchases/common/Backend.kt:61]: it's fine for now, but we really should try to make this a common thing, which we can reuse easily. Perhaps at the HTTPClient level

**Review by aboedo — APPROVED** [review]

File: common/src/main/java/com/revenuecat/purchases/common/Backend.kt (hunk 1 above)
Comment (vegaro) [reply common/src/main/java/com/revenuecat/purchases/common/Backend.kt:61]: Same thoughts I had

File: common/src/main/java/com/revenuecat/purchases/common/Backend.kt [hunk 2]
```diff
@@ -280,7 +291,11 @@ class Backend(
         onSuccessHandler: () -> Unit,
         onErrorHandler: (PurchasesError) -> Unit
     ) {
-        enqueue(object : Dispatcher.AsyncCall() {
+        val cacheKey = listOfNotNull(
+            appUserID,
```
Comment (beylmk) [inline common/src/main/java/com/revenuecat/purchases/common/Backend.kt:295]: looks like tests only check that we don't repeat the call if both are the same. maybe we can add a test to confirm that if EITHER parameter has changed, we WILL repeat the call?

Comment (vegaro) [reply common/src/main/java/com/revenuecat/purchases/common/Backend.kt:295]: Sure, that's a good catch. Want me to write it or you take care of it?

Comment (beylmk) [reply common/src/main/java/com/revenuecat/purchases/common/Backend.kt:295]: i can!

**Pull request merged**